packages/big-design/src/components/Checkbox/spec.tsx

```diff
@@ -46,6 +46,23 @@ describe('render Checkbox', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
+  test('with badge props', () => {
+    const { container } = render(
+      <Checkbox
+        badge={{
+          label: 'Badge',
+          variant: 'secondary',
+        }}
+        description="description text"
+        label="Unchecked"
+        name="test-group"
+        onChange={() => null}
+      />,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
```

Comment:
🍹 A snapshot test might be a little over the top here 🤔 Maybe a `expect(getByText('Badge')).toBeInTheDocument()` could be sufficient.